README.md

```diff
@@ -1,28 +1,157 @@
-# Jaeger OpenShift Templates
+# Getting started with the OpenShift template
 
-Support for deploying Jaeger into OpenShift.
+## Development setup
 
-## Install OpenShift
-If you are new to OpenShift these links help you to get it running:
-* https://install.openshift.com/
-* https://github.com/minishift/minishift
-
-## All-in-one template
-Template with in-memory storage with a limited functionality for local testing and development. 
+This template uses an in-memory storage with a limited functionality for local testing and development.
 Do not use this template in production environments.
 
 To directly install everything you need:
 ```bash
-oc process -f https://raw.githubusercontent.com/jaegertracing/jaeger-openshift/<version>/all-in-one/jaeger-all-in-one-template.yml | oc create -f -
-oc delete all,template -l jaeger-infra    # to remove everything
+oc process -f https://raw.githubusercontent.com/jaegertracing/jaeger-openshift/master/all-in-one/jaeger-all-in-one-template.yml | oc create -f -
 ```
 
-## Troubleshooting
-Tracer sometimes fails to resolve Jaeger's address. In such a case run:
+To cleanup and remove everything, just run:
 ```bash
-sudo iptables -F
+oc delete all,template,daemonset -l jaeger-infra
+```
+
+## Production setup
+
+### TL;DR
+
+Install with:
+```bash
+oc login -u system:admin
+oc new-project jaeger-infra
+oc process -f production/template.yml | oc create -f -
+oc policy add-role-to-user cluster-admin developer -n jaeger-infra
+```
+
+Your Agent hostname is `jaeger-agent.jaeger-infra.svc.cluster.local`
+
+Clean up with:
+```bash
+oc delete all,template,daemonset -l jaeger-infra
+```
+
+### Detailed instructions
+
+You may choose to deploy the Jaeger instance on its own namespace or on an existing namespace,
+with the installation done by an admin or by an unprivileged user. In any case, you'll need your
+Tracer to be pointed to the Agent, which is available at a hostname following this pattern:
+
+```
+jaeger-agent.${NAMESPACE}.svc.cluster.local
+```
+
+For instance, when deployed on its own namespace `jaeger-infra`, this is the full hostname:
+
+```
+jaeger-agent.jaeger-infra.svc.cluster.local
+```
+
+#### Creating the deployment
+
+The `template.yml` file provides a quick start to deploy a production-ready instance of Jaeger on OpenShift.
+The template includes a `StatefulSet` for Cassandra, a `DaemonSet` for the Agent, a `Job` that creates the schema,
+as well as regular services for the individual components such as `Collector` and `Query`. Each one of those
+can be managed and scaled individually.
+
+#### Installing Jaeger on its own namespace as `admin`
+
+Ideally, the Jaeger components would run on its own namespace. For the recommended setup, these are the commands
+to be issued:
+
+```bash
+oc login -u system:admin
```

Comment:
This is inside a section named "... as admin". There's another one, for "... as developer". 
For the Agent to work as a Daemon Set, it does need to be installed as admin (or at least, privileged user), as the daemon set runs at the node level.

Perhaps we could provide another template, for environments like openshift.io , where the Agent won't run as a daemon set?